engine/src/main/scala/cromwell/engine/db/slick/ExecutionComponent.scala

```diff
@@ -167,39 +170,34 @@ trait ExecutionComponent {
       if execution.executionHash === executionHash && execution.allowsResultReuse
     } yield execution)
 
-  /** Returns a tuple of (execution, cache hit workflow UUID, cache hit call FQN)
-    *
-    * If the execution was not satisfied by a cache hit, then the second two values in the tuple are None.
-    *
-    * The query that is being created here is:
-    *
-    * SELECT
-    *     e.*
-    *     we1.WORKFLOW_EXECUTION_UUID as cache_hit_workflow,
-    *     e1.CALL_FQN as cache_hit_call
-    * FROM execution e
-    *     LEFT JOIN execution e1 ON e.results_cloned_from=e1.execution_id
-    *     LEFT JOIN workflow_execution we1 ON e1.workflow_execution_id=we1.workflow_execution_id
-    *     LEFT JOIN workflow_execution we ON e.workflow_execution_id=we.workflow_execution_id
-    * WHERE we.workflow_execution_uuid = 'UUID';
-    *
-    * See the following resources for how to construct queries like this in Slick
-    *
-    *   - http://stackoverflow.com/questions/32061552/slick-3-multiple-outer-joins/32062677
-    *   - http://slick.typesafe.com/doc/3.1.1/queries.html
-    */
-  val executionsWithCacheHitWorkflowAndCall = Compiled(
-    (workflowExecutionUuid: Rep[String]) => for {
-      (((execution, cacheHitExecution), cacheHitWorkflow), workflowExecution) <- executions joinLeft executions on {
-        case (e, cacheHitExecution) => e.resultsClonedFrom === cacheHitExecution.executionId
-      } joinLeft workflowExecutions on {
-        case ((e, maybeCacheHitExecution), we1) => maybeCacheHitExecution.map(_.workflowExecutionId) === we1.workflowExecutionId
-      } joinLeft workflowExecutions on {
-        case (((e, _), _), we) => e.workflowExecutionId === we.workflowExecutionId
-      }
-      if workflowExecution.map(_.workflowExecutionUuid) === workflowExecutionUuid
-    } yield (execution, cacheHitWorkflow.map(_.workflowExecutionUuid), cacheHitExecution.map(_.callFqn))
-  )
+  import slick.dbio.DBIOAction
+
+  def executionsWithCacheHitWorkflowAndCall(workflowExecutionUuid: String): DBIOAction[Vector[ExecutionWithCacheData], NoStream, Read] = {
+    implicit val converter = GetResult { r =>
+      val execution = Execution(r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<)
```

Comment:
haha!  I stole this idea from [the Rawls codebase](https://github.com/broadinstitute/rawls/blob/develop/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala#L51)

I'm definitely not opposed to a better way to do it!